public/apps/configuration/panels/audit-logging/edit-setting-group.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiCodeBlock,
+  EuiComboBox,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTitle,
+} from '@elastic/eui';
+import { get } from 'lodash';
+import { displayBoolean } from '../../utils/display-utils';
+import { comboBoxOptionToString, stringToComboBoxOption } from '../../utils/combo-box-utils';
+
+export function EditSettingGroup(props: {
+  settingGroup: object;
+  config: object;
+  handleChange: (setting: object, val: string) => void;
+  handleInvalid: (setting: object, val: string) => void;
+}) {
+  const renderField = (
+    config: object,
+    setting: object,
+    handleChange: () => void,
+    handleInvalid: () => void
```

Comment:
Please have type definitions for these objects.

Reply:
see above